### PR TITLE
fix: invalid cal video bug

### LINKED ISSUE #12971: [CAL-2850] After reschedule Cal Video link is broken
Consistenly replicable after a reschedule

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2850](https://linear.app/calcom/issue/CAL-2850/after-reschedule-cal-video-link-is-broken)</sub>

### LINKED ISSUE #13060: [CAL-2877] Invalid Cal Video link on calendar event
I had a meeting today that appears in my Google Calendar and in [Cal.com](http://Cal.com)
But when I clicked on the calendar link it redirected me to a [app.cal.com](http://app.cal.com) link with:

image

When I clicked on "Join Cal Video" from the [cal.com](http://cal.com) UI it was actually a different link [https://meetco.daily.co/xxxx](https://meetco.daily.co/xxxx) which worked fine but the meeting attendee didn't get that link.

Thanks!

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2877](https://linear.app/calcom/issue/CAL-2877/invalid-cal-video-link-on-calendar-event)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/12971
Fixes https://github.com/calcom/cal.com/issues/13060

To reproduce the bug

1) Create an event type that requires confirmation with cal video as location
2) Book a meeting and then accept the meeting from /bookings/unconfirmed
3) Reschedule the meeting on Incognito tab or another browser
4) confirm the meeting /bookings/unconfirmed and go to the meeting url

you would get the error "room does not exist or not found" because the room has been deleted when you rescheduled.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/pages/booking/[uid].tsx [hunk 1]
```diff
@@ -830,6 +806,29 @@ export default function Success(props: SuccessProps) {
   );
 }
 
+const DisplayLocation = ({
```
Comment (Udit-takkar) [inline apps/web/pages/booking/[uid].tsx:809]: Just created a common component

File: apps/web/pages/video/[uid].tsx [hunk 2]
```diff
@@ -333,11 +333,14 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
     });
   }
 
+  const videoReferences = bookingObj.references.filter((reference) => reference.type.includes("_video"));
+  const latestVideoReference = videoReferences[videoReferences.length - 1];
+
```
Comment (Udit-takkar) [inline apps/web/pages/video/[uid].tsx:338]: use the latest video reference instead of the first one

**Comment by baileypumfleet** [comment]
Thank you for fixing this, I ran into this issue a few times and it got super annoying

File: apps/web/pages/booking/[uid].tsx (hunk 1 above)
Comment (PeerRich) [reply apps/web/pages/booking/[uid].tsx:809]: smart

**Review by PeerRich — APPROVED** [review]

**Comment by PeerRich** [comment]
can we have E2E tests for this?

**Pull request merged**